### PR TITLE
DM-46387: Use Python logging in ctrl_execute

### PR DESCRIPTION
**Pull request by mxk62**
I replaced all `print()` calls with appropriate logging statements. I deliberately left out `print()` calls in `Allocator.printNodeSetInfo()` as I assume it is supposed to always print a summary information to the console. If it should use logging instead, let me know in the comments.

### CODE REVIEW

File: python/lsst/ctrl/execute/slurmPlugin.py [hunk 1]
```diff
@@ -395,41 +392,41 @@ def glideinsFromJobPressure(self, platformPkgDir):
                 shash = hash[:6]
                 jobname = f"{auser}_{shash}"
                 if verbose:
```
Comment (MichelleGower) [inline python/lsst/ctrl/execute/slurmPlugin.py]: Can be removed like the others?

File: python/lsst/ctrl/execute/slurmPlugin.py [hunk 2]
```diff
@@ -475,45 +469,40 @@ def glideinsFromJobPressure(self, platformPkgDir):
...
+                "small: Number of Glideins to submit is %d", numberOfGlideinsReduced
+            )
 
             cpuopt = f"--cpus-per-task {autoCPUs}"
             memopt = f"--mem {memoryLimit}"
             jobopt = f"-J {jobname}"
             cmd = f"sbatch {cpuopt} {memopt} {jobopt} {generatedSlurmFile}"
-            if verbose:
-                print(cmd)
+            logger.debug("%s", cmd)
```
Comment (MichelleGower) [inline python/lsst/ctrl/execute/slurmPlugin.py]: Can't this be simplified to:  logger.debug(cmd)

File: python/lsst/ctrl/execute/slurmPlugin.py [hunk 3]
```diff
@@ -446,26 +443,23 @@ def glideinsFromJobPressure(self, platformPkgDir):
                 #    print("Using RequestMemoryEval")
```
Comment (MichelleGower) [inline python/lsst/ctrl/execute/slurmPlugin.py]: Go ahead and change this one as well?

**Review by daues — APPROVED** [review]
Tested the changes in multiple scenarios; changes approved.

**Review by MichelleGower — APPROVED** [review]
Just a couple tiny things.  Merge approved.

File: python/lsst/ctrl/execute/slurmPlugin.py (hunk 1 above)
Comment (mxk62) [reply python/lsst/ctrl/execute/slurmPlugin.py]: Good catch! I must have forgotten removing the if statement after changing `print()` to log statement.

**Pull request merged**